ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java

```diff
@@ -196,9 +202,27 @@ protected void storeWindowHandle() {
     }
 
     protected void switchToNewWindow() {
-        for (String winHandle : driver.getWindowHandles()) {
-            driver.switchTo().window(winHandle);
+        Wait<WebDriver> wait = Locator.getFluentWait();
+        wait.until((new Function<WebDriver, Boolean>() {
+            @Override
+            public Boolean apply(WebDriver driver) {
+                try {
+                    // wait for one more window to be opened
+                    return driver.getWindowHandles().size() > 1;
+                } catch (StaleElementReferenceException e) {
+                    return null;
+                }
+            }
+        }));
+        // select last handle
+        String handle = null;
+        Iterator<String> handleIt = driver.getWindowHandles().iterator();
+        while (handleIt.hasNext()) {
+            handle = handleIt.next();
```

Comment:
You can use

    for (String h : driver.getWindowHandles()) {
        handle = h;
    }

Or if you feel fancy

    String handle = driver.getWindowHandles().stream().reduce((prev, next) -> next).orElse(null);

Reply:
:+1: i won't be fancy for clarity :)